assets/helpers/visitToken/reducer.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+// ----- Imports ----- //
+
+import type { Action } from './actions';
+
+
+// ----- Types ----- //
+
+export type VisitToken = {
+    token: ?string,
+};
+
+
+// ----- Setup ----- //
+
+const initialState: VisitToken = window.guardian && window.guardian.visitToken ? {
+  token: window.guardian.visitToken,
+} : {
+  token: null,
+};
+
+
+// ----- Reducer ----- //
+
+export default function visitTokenReducer(
+  state: VisitToken = initialState,
+  action: Action,
+): VisitToken {
+
+  switch (action.type) {
+
+    case 'SET_VISIT_TOKEN':
+      return Object.assign({}, state, { token: action.name });
```

Comment:
This could also be achieved with [object spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals):

```jsx
return { ...state, token: action.name };
```

Reply:
I'm confused now, it seems like the initial state is only used in this function, however the SET_VISIT_TOKEN event is never raised.  However as a side effect of any unrelated event whatsoever being raised, we return the initial state.  This means that we are relying on an arbitrary event being raised in order to fill in the default.  This doesn't seem very accurate design, unless I'm missing something?

Reply:
Are you saying that this state never changes?

Reply:
discussed in person, the answer is yes it never changes, and I've fixed it all in the next PR https://github.com/guardian/support-frontend/pull/1049 😉